test/remittance.js

```diff
@@ -4,7 +4,267 @@ const Remittance = artifacts.require("./Remittance.sol");
 contract('Remittance', async accounts => {
 
     const { toBN } = web3.utils;
+    const concatPassword = "password1" + "password2";
+    const hash = await web3.utils.soliditySha3(concatPassword);
```

Comment:
So a sender that wants to use something other than Javascript will first have to recreate the `utils.soliditySha3` function in this other language?

Reply:
> So a sender that wants to use something other than Javascript will first have to recreate the `utils.soliditySha3` function in this other language?

Good question. Some assumptions I made:

1. That the interactions would be typically via a JS based UI using web3.js
2. That the equivalent function is available in other languages. For example Web3.soliditySha3 in web3py

My idea to initially hash 'off chain' came from this answer:

https://ethereum.stackexchange.com/a/19405/48189

Reply:
Yes, hashing off-chain is important. We should see later that there are conceptually 2 ways of hashing off-chain. But let me rephrase my question.
For someone who wants to recreate `soliditySha3` in C#, how would they know they got it right without having to send countless remittances?

Reply:
> For someone who wants to recreate `soliditySha3` in C#, how would they know they got it right without having to send countless remittances?

I understand. I need to somehow validate that the hash sent has created with the correct hashing algorithm.

Reply:
> For someone who wants to recreate `soliditySha3` in C#, how would they know they got it right without having to send countless remittances?

I have a possible solution:

- The funder also sends a hash of their address when calling .create()
- The create() function hashes the msg.sender and compares it with that hash.

Could that work?

Reply:
Interesting idea, I had not thought about it.
To me this is just a partial solution because you only prevent mistakes. What I would like you to do, is help the `soliditySha3` C# developer create that function.